src/NetAnalyzers/VisualBasic/Microsoft.NetCore.Analyzers/Performance/BasicDoNotUseCountWhenAnyCanBeUsedFixer.vb

```diff
@@ -133,18 +133,25 @@ Namespace Microsoft.NetCore.VisualBasic.Analyzers.Performance
 
             End If
 
-            If invocationExpression Is Nothing Then
+            If invocationExpression IsNot Nothing Then
 
-                expression = Nothing
-                arguments = Nothing
-                Return
+                expression = DirectCast(invocationExpression.Expression, MemberAccessExpressionSyntax).Expression
+                arguments = invocationExpression.ArgumentList.ChildNodes()
+                Return True
 
             End If
 
-            expression = DirectCast(invocationExpression.Expression, MemberAccessExpressionSyntax).Expression
-            arguments = invocationExpression.ArgumentList.ChildNodes()
+            Dim memberAccessExpression = TryCast(sourceExpression, MemberAccessExpressionSyntax)
+            If memberAccessExpression IsNot Nothing Then ' This case happens for something like: x = GetData().Count where Count method is called without parentheses.
```

Comment:
There shall be a similar logic on line 126 (inside of the ` If isAsync Then` block)

Reply:
Correct, that's why `Async` scenarios were not working.

Instead of duplicating the code on the async case, you can refactor the conditions to something like this:
```vb
' Extract source expression from Async expression if needed.
If isAsync Then

    Dim awaitExpressionSyntax = TryCast(sourceExpression, AwaitExpressionSyntax)

    If Not awaitExpressionSyntax Is Nothing Then
        sourceExpression = awaitExpressionSyntax.Expression
    End If

End If

' Try to get source expression as InvocationExpression.
Dim invocationExpression As InvocationExpressionSyntax = TryCast(sourceExpression, InvocationExpressionSyntax)

If invocationExpression IsNot Nothing Then

    expression = DirectCast(invocationExpression.Expression, MemberAccessExpressionSyntax).Expression
    arguments = invocationExpression.ArgumentList.ChildNodes()
    Return True

End If

'  Try to get source expression as MemberAccessExpression which is only legal on VB.
Dim memberAccessExpression = TryCast(sourceExpression, MemberAccessExpressionSyntax)

If memberAccessExpression IsNot Nothing Then
    expression = memberAccessExpression.Expression
    arguments = Enumerable.Empty(Of SyntaxNode)()
    Return True
End If
```